src/framework/audio/internal/synthesizers/fluidsynth/fluidresolver.cpp

```diff
@@ -72,27 +72,62 @@ AudioResourceMetaList FluidResolver::resolveResources() const
     result.reserve(m_resourcesCache.size());
 
     for (const auto& pair : m_resourcesCache) {
-        AudioResourceMeta meta;
-        meta.id = pair.first;
-        meta.type = AudioResourceType::FluidSoundfont;
-        meta.vendor = FLUID_VENDOR_NAME;
-        meta.attributes = { { u"playbackSetupData", mpe::GENERIC_SETUP_DATA_STRING } };
-        meta.hasNativeEditorSupport = false;
-
-        result.push_back(std::move(meta));
+        result.push_back(pair.second.meta);
     }
 
     return result;
 }
 
+static std::string makeId(const std::string& name, int bank, int program)
+{
+    return name + "\\" + std::to_string(bank) + "\\" + std::to_string(program);
+}
+
 void FluidResolver::refresh()
 {
     ONLY_AUDIO_WORKER_THREAD;
 
     m_resourcesCache.clear();
 
-    for (const SoundFontPath& path : soundFontRepository()->soundFontPaths()) {
-        m_resourcesCache.emplace(io::completeBasename(path).toStdString(), path);
+    for (const auto& pair : soundFontRepository()->soundFonts()) {
+        const SoundFontMeta& soundFont = pair.second;
+
+        std::string name = io::completeBasename(soundFont.path).toStdString();
+
+        {
+            AudioResourceId id = name;
+
+            AudioResourceMeta chooseAutomaticMeta;
+            chooseAutomaticMeta.id = id;
+            chooseAutomaticMeta.type = AudioResourceType::FluidSoundfont;
+            chooseAutomaticMeta.vendor = FLUID_VENDOR_NAME;
+            chooseAutomaticMeta.attributes = {
+                { u"playbackSetupData", mpe::GENERIC_SETUP_DATA_STRING },
+                { u"soundFontName", String::fromStdString(name) }
```

Comment:
It's better to declare global static strings and use them instead of raw strings